content/hardware/04.pro/boards/portenta-x8/tutorials/image-building/content.md

```diff
@@ -0,0 +1,211 @@
+---
+beta: true
+title: 'How To Build a Custom Image for Your Portenta X8'
+description: 'This tutorial teaches you how to compile a custom image for your Portenta X8 through USB'
+difficulty: intermediate
+tags:
+  - Embedded Linux
+  - Building
+  - Foundries.io
+author: 'Pablo Marquínez'
+hardware:
+  - hardware/04.pro/boards/portenta-x8
+---
+
+## Overview
+
+In this tutorial you will learn how to build an image for the Portenta X8 with the files provided from Foundries.io.
+
+***In this tutorial we are using Ubuntu LTS 22.04 inside the Windows Subsystem for Linux (WSL) through the Windows Terminal***
+
+## Goals
+
+- Provide your Linux machine with the needed software
+- Set up your computer with the required SSH keys and API Tokens
+- Get the required files
+- Configure the building settings
+- Build the image
+
+### Required Hardware and Software
+
+- [Arduino Portenta X8](https://store.arduino.cc/portenta-x8)
+- Linux distribution [compatible with the Yocto Project](https://docs.yoctoproject.org/ref-manual/system-requirements.html#supported-linux-distributions)
+- Arduino Create account [Login](https://login.arduino.cc/login)
+- Arduino Pro Cloud Subscription. [Learn more about the Pro Cloud](https://www.arduino.cc/pro/hardware/product/portenta-x8#pro-cloud).
+- Foundries.io account (linked with the Pro Cloud subscription) [Login](https://app.foundries.io/login/)
+- FoundriesFactory® ([Check the Getting Started tutorial](https://docs.arduino.cc/tutorials/portenta-x8/out-of-the-box))
+
+## Instructions
+
+### Folder Structure Overview
+
+First we need to structure our directories as following:
+
+```
+home
+└── builds
+  ├── portenta-x8
+  └── .bin
+    └── repo
+```
+
+To start, lets navigate to the home directory on our machine with `cd ~`, and create the build directory with `mkdir builds`.
+
+### Setup SSH Keys and API Token
+
+Next to get the needed image files, we need to setup our SSH Keys and Foundries API token from **FoundriesFactory**.
+
+To setup your SSH Key, go to the `.ssh` directory, `cd ~/.ssh`. Crete a new key with `ssh-keygen -t rsa -b 4096 -C <yourEmail>"`.
+
+Follow the instructions, after which you should have two new files in the directory: `<yourSSHkey> and <yourSSHkey>.pub`
+
+Now initialize the `ssh-agent` with `eval "$(ssh-agent -s)"`. Add your key with `ssh-add ~/.ssh/<yourSSHkey>`.
+
+***Make sure your `ssh-agent` is running, you can check the keys that your agent has with the command `ssh-add -l` if there is no key attached repeat the steps of adding the key and check if it was successfully added***
```

Comment:
I'm not sure if this section is really needed, can you double check?
```suggestion
### Setup SSH Keys and API Token

Next to get the needed image files, we need to setup our SSH Keys and Foundries API token from **FoundriesFactory**.

To setup your SSH Key, go to the `.ssh` directory, `cd ~/.ssh`. Crete a new key with `ssh-keygen -t rsa -b 4096 -C <yourEmail>"`.

Follow the instructions, after which you should have two new files in the directory: `<yourSSHkey> and <yourSSHkey>.pub`

Now initialize the `ssh-agent` with `eval "$(ssh-agent -s)"`. Add your key with `ssh-add ~/.ssh/<yourSSHkey>`.

***Make sure your `ssh-agent` is running, you can check the keys that your agent has with the command `ssh-add -l` if there is no key attached repeat the steps of adding the key and check if it was successfully added***
```

Reply:
I needed the ssh key for fetching some repos when bitbake

The API Token needs to be there in order to access the factory source code

Reply:
not needed